lib/takeaway_service.rb

```diff
@@ -0,0 +1,92 @@
+require_relative 'menu'
+require_relative 'dish'
+require_relative 'send_sms'
+
+class TakeawayService
+  attr_reader :basket
+
```

Comment:
I like the use of dependency injection with the menu_class. I don't think you need to add the text_service_class to the initialize of the TakeawayService class.